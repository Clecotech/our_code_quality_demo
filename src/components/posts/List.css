.Table {
  width: 100%;
  margin-top: 1.5rem;
  background-color: #f2f2f2;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}
.PostContainer {
  display: flex;
  align-items: center;
  justify-content: center;
}
.Postbtn {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
}
