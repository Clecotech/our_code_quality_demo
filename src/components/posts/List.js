import React, { Component } from "react";
import PropTypes from "prop-types";
import { Button, Grid } from "@mui/material";
import "./List.css";
export class List extends Component {
  static propTypes = {
    index: PropTypes.number,
    title: PropTypes.string,
    body: PropTypes.string,
    onPostDeleted: PropTypes.func,
  };

  state = {
    showTitleBackwards: false,
  };

  onTitleClicked = () =>
    this.setState({
      showTitleBackwards: !this.state.showTitleBackwards,
    });

  onDeletePostClick = () =>
    this.props.onPostDeleted && this.props.onPostDeleted(this.props.index);

  render() {
    const { title, body } = this.props;
    const { showTitleBackwards } = this.state;
    return (
      <div className="Table">
        <Grid container>
          <Grid item xs={2} sm={4}>
            <div className="PostContainer">
              {showTitleBackwards ? title.split("").reverse().join("") : title}
            </div>
          </Grid>
          <Grid item xs={2} sm={4}>
            <div className="PostContainer">{body}</div>
          </Grid>
          <Grid item xs={8} sm={4} className="PostContainer">
            <div className="Postbtn">
              <Button
                id="reverse"
                onClick={this.onTitleClicked}
                variant="contained"
                sx={{ borderRadius: "1.5rem" }}
              >
                Reverse Title
              </Button>

              <Button
                id="delete"
                onClick={this.onDeletePostClick}
                variant="outlined"
                sx={{ borderRadius: "1.5rem" }}
              >
                Delete Post
              </Button>
            </div>
          </Grid>
        </Grid>
      </div>
    );
  }
}
