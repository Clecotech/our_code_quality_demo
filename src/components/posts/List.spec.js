import React from "react";
import { shallow } from "enzyme";
import { List } from "./List";

describe("Post", () => {
  const examplePost = {
    index: 0,
    title: "John",
    body: "Doe",
  };

  it("renders a post's title and body", () => {
    const wrapper = shallow(<Post {...examplePost} />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("delete button clicked", () => {
    it("fires onPostDeleted event", () => {
      const mockedEvent = jest.fn();
      const wrapper = shallow(
        <Post {...examplePost} onPostDeleted={mockedEvent} />
      );
      wrapper.find("#delete").simulate("click");
      expect(mockedEvent.mock.calls.length).toBe(1);
    });
  });
});
