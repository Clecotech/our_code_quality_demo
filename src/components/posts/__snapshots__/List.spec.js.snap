// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders a post's title and body 1`] = `
<div
  className="Table"
>
  <ForwardRef(Grid)
    container={true}
  >
    <ForwardRef(Grid)
      item={true}
      sm={4}
      xs={2}
    >
      <div
        className="PostContainer"
      >
        John
      </div>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      sm={4}
      xs={2}
    >
      <div
        className="PostContainer"
      >
        Doe
      </div>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      className="PostContainer"
      item={true}
      sm={4}
      xs={8}
    >
      <div
        className="Postbtn"
      >
        <ForwardRef(Button)
          id="reverse"
          onClick={[Function]}
          sx={
            Object {
              "borderRadius": "1.5rem",
            }
          }
          variant="contained"
        >
          Reverse Title
        </ForwardRef(Button)>
        <ForwardRef(Button)
          id="delete"
          onClick={[Function]}
          sx={
            Object {
              "borderRadius": "1.5rem",
            }
          }
          variant="outlined"
        >
          Delete Post
        </ForwardRef(Button)>
      </div>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;
